### PR TITLE
wsd: new NetUtil file for network utilities

### PR DESCRIPTION
**Pull request by Ashod**
Move the connect function into the NetUtil
translation unit to aid using it for the
upcoming async socket logic.

The NetUtil should also come in handy for
the miscellaneous network helpers we have.

Change-Id: I2ee0c6e3e1769fd87572d7407d3b4979b59ffe6a
Signed-off-by: Ashod Nakashian <ashod.nakashian@collabora.co.uk>

### CODE REVIEW

File: net/NetUtil.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+#if ENABLE_SSL && !MOBILEAPP
+#include "SslSocket.hpp"
+#endif
+
+#include <netdb.h>
+
+namespace net
+{
+std::shared_ptr<StreamSocket>
+connect(const std::string& host, const std::string& port, const bool encrypted,
```
Comment (kendy) [inline net/NetUtil.cpp:22]: Would be great not to rename the "isSSL" to "encrypted" if possible - "encrypted" is too general in my view, making the code reading a bit harder (at least for me) :-)  Can we keep "isSSL" (or "useSSL" if that does not fit) please?

File: test/Makefile.am [hunk 2]
```diff
@@ -89,6 +89,7 @@ unittest_SOURCES = \
     ../common/Unit.cpp \
     ../common/StringVector.cpp \
     ../net/Socket.cpp \
+	../net/NetUtil.cpp \
```
Comment (kendy) [inline test/Makefile.am:92]: Cosmetic - indentation...

**Review by kendy — COMMENTED** [review]
@Ashod: Thanks for that!  Looks good to me in general, commented about some small things; would be good to fix if possible, otherwise good to go in from my point of view.  Thank you!

File: net/NetUtil.cpp (hunk 1 above)
Comment (Ashod) [reply net/NetUtil.cpp]: Yes, no good reason to change it. I typed `encrypted` separately, but now changed to isSSL.

File: test/Makefile.am (hunk 2 above)
Comment (Ashod) [reply test/Makefile.am]: Ironic that I had noticed all the tabs and made sure (failed, of course) to use spaces (I blame my editor). I replaced the tabs to spaces.

**Pull request merged**